backend/app/Http/Requests/AbstractChartHttpRequest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Requests;
+
+use App\Http\Request\ApiFormRequest;
+use App\Rules\Timestamp;
+use App\Rules\TimestampAfter;
+
+abstract class AbstractChartHttpRequest extends ApiFormRequest
```

Comment:
I like your idea to move general logic to some abstraction. But instead of inheritance, it is much better to use traits